repos/config/detail/generate_ros2_config.py

```diff
@@ -35,13 +36,33 @@ def print_setup(repos):
 
 
 def build_load_command(repo, spec):
-    if spec['type'] == "git":
-        return build_remote_load_command(repo, spec)
-    if spec['type'] == "local":
+    if spec.get('type') == "git":
+        return build_git_load_command(repo, spec)
+    if spec.get('type') == "http_archive":
+        return build_http_archive_load_command(repo, spec)
+    if spec.get('type') == "local":
         return build_local_load_command(repo, spec)
     else:
-        raise ValueError(f"Unknown repo type {spec['type']}")
+        return f"""
+    print("WARNING: Unknown repo type {spec.get('type')} for repo @{repo.replace('/', '.')}")
+"""
```

Comment:
@kilian-funk the formatting looks a little funky here, is there a `dedent` function in Starlark similar to that of python's `textwrap.dedent` method?

Reply:
@flynneva This is Python. So ,any suggestion to beautify this is welcome. I have not worked with the textwrap module so far.

Reply:
@kilian-funk ah! haha my bad! Yes in python I like the `dedent` method a lot, since it makes this sort of multi-line strings much easier to read without affecting the indentation of the final string.

```suggestion
        from textwrap import dedent

        return dedent(
            f"""
                print("WARNING: Unknown repo type {spec.get('type')} for repo @{repo.replace('/', '.')}")
            """
        )
```

The `dedent` method removes any indentation before the opening triple quotes `"""` but leaves the rest, so you can format this string to exactly how you want the final string to look without having some wacky looking code. Not so useful here for just a one-liner, but really useful when you start having many-line multistrings to format properly.

Reply:
you can use a backslash also to remove any unwanted new line statements that might pop up from the string, a lot of times the opening triple quotes I follow with a backlash `"""\` to remove the leading new line